src/static/riot/competitions/detail/leaderboards.tag

```diff
@@ -44,114 +44,53 @@
         </tr>
         </tbody>
     </table>
+
     <script>
         let self = this
         self.selected_leaderboard = {}
-        self.hidden_column_keys = []
-        self.filtered_column_keys = new Set()
-        self.competition_id = 0
+        self.columns = []
+        self.phase_id = null
+        self.competition_id = null
 
         self.get_score = function(column, scores) {
-            for (i in scores) {
-                score = scores[i]
-                if (column.key === score.column_key) {
-                    return score.score
-                }
+            let score = _.get(_.find(scores, {'task_id': column.task_id, 'column_key': column.key}), 'score')
+            if (score) {
+                return score
             }
             return 'n/a'
         }
 
         self.update_leaderboard = () => {
-            if (_.isEmpty(self.selected_leaderboard)) {
-                return
-            }
-            self.hidden_column_keys = _.map(self.selected_leaderboard.columns, col => {
-                if (col.hidden) {
-                    return col.key
-                }
-            })
-
-            // Columns are assumed to be the same for each task. Each column on the leaderboard is duplicated once for
-            // each task and assigned a unique column key.
-            self.generated_columns = []
-            _.forEach(self.opts.tasks, task => {
-                _.forEach(self.selected_leaderboard.columns, col => {
-                    col = Object.assign({}, col)
-                    col['task'] =  task
-                    col['key'] += `_${task['id']}`
-                    self.generated_columns.push(col)
-                })
-            })
-
-            // organized_submissions is used to organize submissions by owner so many scores by the same owner are
-            // rendered on the same row of the leaderboard table.
-            let organized_submissions = {}
-            _.forEach(self.selected_leaderboard.submissions, submission => {
-                _.forEach(submission['scores'], score => {
-                    score['column_key'] += `_${submission['task']}`
+            CODALAB.api.get_leaderboard_for_render(self.phase_id)
+                .done(responseData => {
+                    self.selected_leaderboard = responseData
+                    self.columns = []
+                    for(taskNum = 0; taskNum < Object.keys(self.selected_leaderboard.tasks).length; taskNum++){
+                        for(colNum = 0; colNum < Object.keys(self.selected_leaderboard.tasks[taskNum].columns).length; colNum++){
+                            col = self.selected_leaderboard.tasks[taskNum].columns[colNum]
+                            col.task_id = self.selected_leaderboard.tasks[taskNum].id
+                            self.columns.push(col)
+                        }
+                    }
+                    self.update()
```

Comment:
This guy is much cleaner than my mess. Nice.
Also fyi you can do `for(taskNum in self.selected_leaderboard.tasks)` instead of
`for(taskNum = 0; taskNum < Object.keys(self.selected_leaderboard.tasks).length; taskNum++)`
I believe it's functionally the same but I think it looks a little cleaner.